openwisp_firmware_upgrader/tests/test_models.py

```diff
@@ -483,3 +484,33 @@ def test_delete_firmware_image_file(self):
             file_storage_backend.delete(file_name)
             self.assertNotEqual(image.file, None)
             image.delete()
+
+    def test_filename_within_limit(self):
+        build_object = self._create_build()
+        # Created a filename exactly at the limit of 255 characters
+        long_filename = "a" * 255
+
+        dummy_file = SimpleUploadedFile(long_filename, b"file_content")
+        firmware_image = FirmwareImage(
+            build=build_object,
+            file=dummy_file,
+            type='factory',
+        )
+        self.assertIsNotNone(firmware_image.pk)
+        expected_filename = long_filename
+        self.assertEqual(firmware_image.file.name, expected_filename)
```

Comment:
I would still call `fimrware_image.full_clean()` here to ensure that validation is not failing.

Just for the sake of readability, I would wrap that in try-except like so 

```suggestion
        try:
            firmware_image.full_clean()
        except ValidationError: 
            self.fail('Unexpected ValidationError raised on running full_clean')
```

Feel free to improve the fail message.